src/pages/courses/[id].tsx

```diff
@@ -0,0 +1,150 @@
+import { Box, Flex, Heading, Image, Text, AspectRatio } from "@chakra-ui/react";
+import { useNavigate, useParams } from "react-router-dom";
+import CourseData from "../../lib/db/courses.json";
+import { FaChevronLeft } from "react-icons/fa6";
+import ViewportContainer from "../../layouts/container";
+import ErrorPage from "../ErrorPage";
+import CustomButton from "../../components/custom-button";
+import CourseListSection from "../../components/course-list-section";
+
+const CourseInfo = () => {
+  const { id } = useParams();
+  const navigate = useNavigate();
+  const courseInfo = CourseData.find((course) => course.slug === id);
+
+  if (!courseInfo) {
+    return <ErrorPage errorTitle="Cannot find requested course" />;
+  }
+
+  return (
+    <Box className="page">
+      <Box position="relative" w="100%" maxH="482px" mb={12}>
+        <Image
+          src={courseInfo.coverImage}
+          alt={courseInfo.title}
+          objectFit="cover"
+          objectPosition={
+            courseInfo.id === "course-001" ? "center" : "top center"
+          }
+          w="100%"
+          h={{ base: "320px", md: "482px" }}
+        />
+        <Box
+          position="absolute"
+          top={0}
+          left={0}
+          w="100%"
+          h="100%"
+          bg="rgba(0, 0, 0, 0.6)"
+          zIndex={2}
+        />
+        <Flex
+          position="absolute"
+          top="50%"
+          left={{ base: 6, md: "8rem" }}
+          transform="translateY(-50%)"
+          zIndex={3}
+          direction="column"
+          justify="center"
+          align="flex-start"
+          h="auto"
+          maxW={{ base: "100%", md: "50%" }}
+        >
+          <CustomButton
+            variant="ghost"
+            color="white"
+            _hover={{ color: "purple.300" }}
+            leftIcon={<FaChevronLeft />}
+            onClick={() => navigate(-1)}
+          >
+            Back
+          </CustomButton>
+          <Heading
+            as="h1"
+            color="white"
+            fontSize={{ base: "heading2", md: "6.5rem" }}
+          >
+            {courseInfo.title}
```

Comment:
```suggestion
            {courseInfo?.title}
```

if this is coming from an API, it would break because course would first be `undefined`